Assets/BossRoom/Scripts/Game/Server/Action/PickUpAction.cs

```diff
@@ -0,0 +1,123 @@
+using Unity.Multiplayer.Samples.BossRoom.Client;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Animations;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Server
+{
+    /// <summary>
+    /// Action for picking up "Heavy" items. For simplicity, this class will perform both the pickup (reparenting) of a
+    /// NetworkObject, as well as the drop (deparenting).
+    /// </summary>
+    public class PickUpAction : Action
+    {
+        RaycastHit[] m_RaycastHits = new RaycastHit[1];
+
+        const string k_HeavyTag = "Heavy";
+
+        const string k_NpcLayer = "NPCs";
+        public PickUpAction(ServerCharacter parent, ref ActionRequestData data) : base(parent, ref data)
+        {
+        }
+
+        public override bool Start()
+        {
+            var pickUpObject = m_Parent.GetComponentInChildren<NetworkPickUpState>();
+
+            // first, check if a pot has already been parented; if so, drop it
+            if (pickUpObject)
+            {
+                pickUpObject.transform.SetParent(null);
+
+                Data.TargetIds = null;
+
+                if (!string.IsNullOrEmpty(Description.Anim))
+                {
+                    m_Parent.serverAnimationHandler.NetworkAnimator.ResetTrigger(Description.Anim);
+                }
+
+                // drop
+                if (!string.IsNullOrEmpty(Description.Anim2))
+                {
+                    m_Parent.serverAnimationHandler.NetworkAnimator.SetTrigger(Description.Anim2);
+                }
+
+                return false;
+            }
+
+            var numResults = Physics.BoxCastNonAlloc(m_Parent.physicsWrapper.Transform.position,
+                m_Parent.physicsWrapper.DamageCollider.bounds.extents,
+                m_Parent.physicsWrapper.Transform.forward,
+                m_RaycastHits,
+                Quaternion.identity,
+                Description.Range,
+                1 << LayerMask.NameToLayer(k_NpcLayer));
+
+            // collider must contain "Heavy" tag
+            if (numResults == 0 || !m_RaycastHits[0].collider.TryGetComponent(out NetworkObject heavyNetworkObject) ||
+                !m_RaycastHits[0].collider.gameObject.CompareTag(k_HeavyTag))
+            {
+                return false;
+            }
+
+            // found a suitable collider; make sure it is not already held by another player
+            if (heavyNetworkObject.transform.parent != null &&
+                heavyNetworkObject.transform.parent.TryGetComponent(out NetworkObject parentNetworkObject))
+            {
+                // pot already parented; return for now
+                return false;
+            }
+
+            // found a suitable collider; try to child this NetworkObject
+            if (!heavyNetworkObject.TrySetParent(m_Parent.transform))
```

Comment:
If you add artificial jitter, do you see issues where the pot doesn't quite follow the character's hands when walking and moving it?
A solution for that is to use local positions (that's an option you can set in NetworkTransform)

Reply:
It's the same jitter that a player would experience since the pot is synced via NetworkTransform as well.

Reply:
As discussed, using local pos is also a good way to reduce bandwidth consumption (doc for this should be created)

Reply:
Is there a ticket for doc writing on this? Or a PR? Just so we don't lose track of this

Reply:
Here: [MTT-4466](https://jira.unity3d.com/browse/MTT-4466).